### PR TITLE
Feature NXP-19171 better error message for ressource bundle configuration

### PR DESCRIPTION
**Pull request by ChevalierFrugier**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/nuxeo/nuxeo/pull/163" target="_blank">ReviewNinja

File: nuxeo-services/nuxeo-web-resources-core/src/test/java/org/nuxeo/ecm/web/resources/tests/TestWebResourceService.java [hunk 1]
```diff
@@ -127,6 +149,7 @@ public void testResourceBundle() throws Exception {
         String contrib = "webresources-test-override-config.xml";
         URL url = getClass().getClassLoader().getResource(contrib);
         try (RuntimeContext ctx = harness.deployTestContrib(BUNDLE, url)) {
+            logCaptureResult.assertHasEvent();
```
Comment (kevinleturc) [inline nuxeo-services/nuxeo-web-resources-core/src/test/java/org/nuxeo/ecm/web/resources/tests/TestWebResourceService.java]: Maybe you can assert the message ?

File: nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java [hunk 2]
```diff
@@ -99,6 +111,10 @@ public ResourceBundle merge(ResourceBundle other) {
             if (res != null) {
                 merged.addAll(res);
             }
+            if (merged.removeAll(Arrays.asList("", null))) {
+                log.error("Some resources references were null or blank while setting " + this.name
+                    + " and have been supressed. This probably happened because some <resource> tags were empty in the xml declaration. The correct form is <resource>resource name</resource>.");
+            }
```
Comment (kevinleturc) [inline nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java]: You the same block of code, you can put this in a private method.
Furthermore I'm not sure if the case is possible in merge method as all resources set seem to call setResources.

File: nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java [hunk 3]
```diff
@@ -35,19 +38,23 @@
...
+
     @XNode("@name")
     public String name;
 
     @XNode("resources@append")
     boolean append;
 
     @XNodeList(value = "resources/resource", type = ArrayList.class, componentType = String.class)
-    List<String> resources;
+    private List<String> resources;
```
Comment (kevinleturc) [inline nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java]: You can remove this change

**Comment by jcarsique** [comment]
typo: supress => suppress

nice use of `LogCaptureFeature`
maybe that the `BadRessourceDeclarationLogFilter` is generic enough to be factorized into a parameterized annotation for filtering on the given log level and logger name? See `org.nuxeo.ecm.automation.core.TestRestoreInputFromScriptAndLogOperation.MyLogFilter`, `org.nuxeo.ecm.core.test.QueryResultsAreAutomaticallyClosedTest.LogFilter`, `org.nuxeo.ecm.core.io.download.TestDownloadService.DownloadServiceEtagNoDigestFilter`
=> if you think so, just create a dedicated JIRA ticket; no need for it in your commit

File: nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java (hunk 3 above)
Comment (ChevalierFrugier) [inline nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java]: !completed

File: nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java (hunk 2 above)
Comment (ChevalierFrugier) [inline nuxeo-services/nuxeo-web-resources-core/src/main/java/org/nuxeo/ecm/web/resources/core/ResourceBundleDescriptor.java]: !completed

**Comment by kevinleturc** [comment]
👍

**Pull request merged**